### PR TITLE
support dynamic shape

### PR DESCRIPTION
**Pull request by yotamazriel**
- fix reshape before dense to assert channels length
- support dynamic shapes if input is keras tensor
- support dyanmic reshape
- support dyanmic slicing using strided slice
- add bert huggingface for classification test
- change resize use dyanmic shapes

### CODE REVIEW

**Review by doronHarnoy — APPROVED** [review]
LGTM

File: onnx2kerastl/reshape_layers.py [hunk 1]
```diff
@@ -372,25 +372,49 @@ def convert_slice(node, params, layers, lambda_func, node_name, keras_name):
...
-    for axis in range(input_shape_len):
-        if axis in axes_positives:
-            axis_index = axes_positives.index(axis)
-            start = starts[axis_index]
-            end = ends[axis_index]
-            step = steps[axis_index]
-            slice_spec_param.append({'start': start, 'step': step, 'stop': end})
+    is_dynamic = False
+    for i in range(len(starts)):
+        for index_li in zip([starts, steps, ends]):
```
[the remark this answers is not in the document]
Comment (tomkoren21) [reply onnx2kerastl/reshape_layers.py]: Solved

File: onnx2kerastl/linear_layers.py [hunk 2]
```diff
@@ -50,8 +50,9 @@ def convert_gemm(node, params, layers, lambda_func, node_name, keras_name):
             try:
                 layers[node_name] = dense(layers[node.input[0]])
             except ValueError:
-                reshape = keras.layers.Reshape([input_channels], name=keras_name + '_reshape')
-                reshaped_x = reshape(layers[node.input[0]])
+                mid_shape = tf.shape(layers[node.input[0]], out_type=tf.int32)[:-1]
+                reshape_shape = tf.concat([mid_shape, [input_channels]], axis=0)
```
[the remark this answers is not in the document]
Comment (tomkoren21) [reply onnx2kerastl/linear_layers.py:54]: This fixes (*,*,None)-like input and turns them into (*,*,num-channels). In essence, it does nothing - but is a must for the model to compile, for dynamic models.

[the remark this answers is not in the document]
Comment (doronHarnoy) [reply onnx2kerastl/linear_layers.py:54]: cool!

**Pull request merged**